env_spec.py

```diff
@@ -19,138 +28,216 @@
 def create_list(input_str):
     """
     Creates a list from input_str_list.Returns the new list.
+    If input_str=[1,2,], returns empty list(syntax error).
     """
     input_str = input_str.replace("[", "")
     input_str = input_str.replace("]", "")
     new_list = input_str.split(",")
 
+    if "" in new_list:
+        return []
+
     for pos in range(0, len(new_list)):
-        new_list[pos] = str(new_list[pos])
+        new_list[pos] = str(new_list[pos]).strip()
 
     return new_list
 
 
-def check_input(input_field):
+def render_label(env_spec_entry):
     """
-    Checks syntax of input field(uppercase latin chars,numeric digits, underscores).
-    Returns False for error or True for correct syntax.
+    Creates the html output for label.Takes as argument the current dictionary.
     """
-    alphanumeric_that_does_not_start_with_digit = r"^[A-Z_][0-9A-Z_]*$"
+    label_str = f'<label for="env_spec_{env_spec_entry["name"].lower()}">{env_spec_entry["name"]}</label>\n'
+
+    return label_str
 
-    ret = re.match(alphanumeric_that_does_not_start_with_digit, input_field)
 
-    if ret is None:
-        return False
+def render_input(env_spec_entry):
+    """
+    Creates the html output for input.Takes as argument the current dictionary.
+    """
+    if env_spec_entry["default_value"] is None:
+        input_str = f'<input id="env_spec_{env_spec_entry["name"].lower()}" name="{env_spec_entry["name"].lower()}" type="{env_spec_entry["type"]}" />\n'
+    else:
+        input_str = f'<input id="env_spec_{env_spec_entry["name"].lower()}" name="{env_spec_entry["name"].lower()}" type="{env_spec_entry["type"]}" value="{env_spec_entry["default_value"]}" />\n'
 
-    return True
+    return input_str
 
 
-def check_type(input_type):
+def render_choice(choice, selected=False):
     """
-    Checks if given type is valid and returns True, else returns False.
+    Creates the html output for key "choice".
     """
+    if selected is False:
```

Comment:
Since most of the text between the two kinds of options (selected and not selected) is the same, I propose changing this to something more Pythonic to [avoid repeating ourselves](
https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).

Something like the following works:

```py
selected_state = "selected" if selected else ""

return f'\t<option value="{choice}" {selected_state}>{choice}</option>\n'
```